p2p/src/types/peer_address.rs

```diff
@@ -35,7 +35,7 @@ pub struct PeerAddressIp6 {
 ///
 /// Same as std::net::SocketAddr for now but can be later extended with other address types.
 /// Use custom type to be able implement Encode and Decode.
-#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq)]
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, Hash)]
```

Comment:
As I always say, please do not use `Hash` trait as it's easily confused with other Hash things in our software. For maps, just use BTreeMap and implement `Ord` instead.

Reply:
This is the first time I have heard about it. But I think banning `Hash` trait from stdlib just because it might be confusing is too rough. Anyway, I removed it because `Hash` is not actually needed currently.

Reply:
This is a discussion I'm more than happy to have with the team. Anyway, I think hash maps are overused anyway and people treat them as silver bullets for performance. It doesn't matter that much and people cannot prove the difference except with benchmarks. Who cares.